### ISSUE TITLE
[Bug]:  mo was hung when running tpch 100g q3 on eks with 3*cn

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9978]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-29f1e83
- Hardware parameters:
  3*CN: 16C 64G
  1*DN: 16C 64G
  3*LOG: 4C 16G
- OS type:
- Others:
  DISK CACHE: 100G
  MEM CACHE: 12G
```

### Actual Behavior

when running tpch 100g q3 on eks with 3*cn, the mo was hung and all the following requests were hung.
<img width="1407" alt="image" src="https://github.com/matrixorigin/matrixone/assets/47846308/d647b42b-283c-4507-8843-042a6e943010">

mo log:
[cn-0.log](https://github.com/matrixorigin/matrixone/files/11713615/cn-0.log)
[cn-1.log](https://github.com/matrixorigin/matrixone/files/11713616/cn-1.log)
[cn-2.log](https://github.com/matrixorigin/matrixone/files/11713617/cn-2.log)

mo profile:
[prof-cn0.tar.gz](https://github.com/matrixorigin/matrixone/files/11713622/prof-cn0.tar.gz)
[prof-cn1.tar.gz](https://github.com/matrixorigin/matrixone/files/11713623/prof-cn1.tar.gz)
[prof-cn2.tar.gz](https://github.com/matrixorigin/matrixone/files/11713625/prof-cn2.tar.gz)

### Steps to Reproduce

```Markdown
1. run a mo cluster with eks with 3*cn
2. run tpch 100g queries
```

**Comment by aressu1985** [comment]
I can not repro this. so It is very hard to confirm which pr cause this

**Comment by sukki37** [comment]
![image](https://github.com/matrixorigin/matrixone/assets/77312370/1c6655c5-57cf-488e-bd6f-aa5af89a6769)
There may be a problem with cn-dn's logtail. cn1 keeps reporting errors. @m-schen please help on identify the root cause of the issue, thank you.

**Comment by m-schen** [comment]
看起来是cn1与dn已经失去连接了。但是由于cn之间还可以ping通，因此把查询发到cn1执行了。
我检查一下为什么没有触发重连等。

**Comment by m-schen** [comment]
> 看起来是cn1与dn已经失去连接了。但是由于cn之间还可以ping通，因此把查询发到cn1执行了。

没有失去连接，在消费某个logtail时卡住了。卡在了
<img width="797" alt="image" src="https://github.com/matrixorigin/matrixone/assets/59043531/2df2ff1c-104d-48c9-8145-0a714ef945df">

**Comment by m-schen** [comment]
https://github.com/matrixorigin/matrixone/pull/10001 修复了这个错误。可以验证。

**Comment by aressu1985** [comment]
fixed: nightly-d4d74e7
